frames/src/test/java/com/checkout/frames/component/expirydate/SmartExpiryDateValidationUseCaseTest.kt

```diff
@@ -0,0 +1,293 @@
+import java.util.stream.Stream
+
+@SuppressLint("NewApi")
+@ExtendWith(MockKExtension::class)
+internal class SmartExpiryDateValidationUseCaseTest {
+
+    @RelaxedMockK
+    lateinit var mockCardValidator: CardValidator
+
+    private lateinit var expiryDateValidationUseCase:
+            UseCase<SmartExpiryDateValidationRequest, ValidationResult<String>>
+
+    private lateinit var fakeExpiryDate: ExpiryDate
+
+    @BeforeEach
+    fun setUp() {
+        expiryDateValidationUseCase = SmartExpiryDateValidationUseCase(mockCardValidator)
+        fakeExpiryDate = ExpiryDate(12, 12)
+    }
+
+    @ParameterizedTest(
+        name = "Expected success validation result {0} received for OnFocus when valid {1} expiry date requested"
+    )
+    @MethodSource("validateSuccessExpiryDateOnFocusBehaviourArguments")
+    fun `when valid expiry date is requested then return success result for On Focus behaviour`(
+        inputRequest: SmartExpiryDateValidationRequest,
+        expectedExpiryDate: ValidationResult<String>,
+    ) {
+        // Given
+        every {
+            mockCardValidator.validateExpiryDate(any() as String, any() as String)
+        } returns ValidationResult.Success(fakeExpiryDate)
+
+        // When
+        val result = expiryDateValidationUseCase.execute(inputRequest)
+
+        // Then
+        when (expectedExpiryDate) {
+            is ValidationResult.Success -> {
+                Assertions.assertEquals(expectedExpiryDate.value, (result as? ValidationResult.Success<String>)?.value)
+            }
+            is ValidationResult.Failure -> Assertions.assertEquals(
+                expectedExpiryDate.error.message,
+                (result as? ValidationResult.Failure)?.error?.message
+            )
+        }
+    }
+
+    @ParameterizedTest(
+        name = "Expected success validation result {0} received for OnFocusChanged when valid {1} expiry date requested"
+    )
+    @MethodSource("validateSuccessExpiryDateOnFocusChangedBehaviourArguments")
+    fun `when valid expiry date is requested then return success result for On Focus Changed behaviour`(
+        inputRequest: SmartExpiryDateValidationRequest,
+        expectedExpiryDate: ValidationResult<String>,
+    ) {
+        // Given
+        every {
+            mockCardValidator.validateExpiryDate(any() as String, any() as String)
+        } returns ValidationResult.Success(fakeExpiryDate)
+
+        // When
+        val result = expiryDateValidationUseCase.execute(inputRequest)
+
+        // Then
+        when (expectedExpiryDate) {
+            is ValidationResult.Success -> {
+                Assertions.assertEquals(expectedExpiryDate.value, (result as? ValidationResult.Success<String>)?.value)
+            }
+            is ValidationResult.Failure -> Assertions.assertEquals(
+                expectedExpiryDate.error.message,
+                (result as? ValidationResult.Failure)?.error?.errorCode
+            )
+        }
+    }
```

Comment:
The only difference is a method source, am I right?
If yes, why not use one test and provide both test sources to reduce code repetition?